rust/syntect-plugin/src/main.rs

```diff
@@ -73,7 +75,8 @@ struct PluginState {
     // unflushed spans
     spans: Vec<ScopeSpan>,
     new_scopes: Vec<Vec<String>>,
-    indentation_state: Vec<(usize, EditType)>,
+    // keeps track of the lines (start, end) that might need indentation after edit
+    indentation_state: Vec<(usize, usize, EditType)>,
```

Comment:
I might start to think about something like,

```rust
enum IndentationTask {
    Newline(usize),
    Edit(usize),
    Batch(Range<usize>),
}
```

Reply:
Looks definitely cleaner